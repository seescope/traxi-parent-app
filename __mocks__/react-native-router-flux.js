import React, { PropTypes } from 'react';
import { View } from 'react-native';

export const Actions = {
  createKid: jest.fn(),
  walkthrough: jest.fn(),
  congratulations: jest.fn(),
  thankyou: jest.fn(),
  areYouReady: jest.fn(),
  intro: jest.fn(),
};

export const Scene = (props) => <View {...props} />;

export const Router = ({ children }) => (
  <View>{children}</View>
);

Router.propTypes = {
  children: PropTypes.array.isRequired,
};
