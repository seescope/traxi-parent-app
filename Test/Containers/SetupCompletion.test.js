import { Alert } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { Actions } from 'react-native-router-flux';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
jest.mock('../../App/AsyncActions/PersistParent', () => jest.fn());
jest.mock('../../App/AsyncActions/FetchReports', () => jest.fn());
jest.mock('../../App/AsyncActions/CreateParentAuthentication', () => jest.fn());
import mockPersistParent from '../../App/AsyncActions/PersistParent';
import mockFetchReports from '../../App/AsyncActions/FetchReports';
import mockCreateParentAuthentication
  from '../../App/AsyncActions/CreateParentAuthentication';

import SetupCompletionComponent, {
  mapDispatchToProps,
} from '../../App/Containers/SetupCompletion';

Alert.alert = jest.fn();

const mockStore = configureStore([thunk]);
const testStore = mockStore({
  kidsState: {
    'abc-123': {
      name: 'John Bobson',
    },
  },
  setupState: {
    kidUUID: 'abc-123',
  },
  parentState: {},
});

const SetupCompletion = () => (
  <Provider store={testStore}>
    <SetupCompletionComponent />
  </Provider>
);

describe('SetupCompletion', () => {
  it('Renders correctly', () => {
    const tree = renderer.create(<SetupCompletion />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('mapDispatchToProps', () => {
  it('onPress handles loading calls createParentAuthentication and persistParent, then goes to dashboard', () => {
    const mockDispatch = jest.fn(() => Promise.resolve());
    const { onPress } = mapDispatchToProps(mockDispatch);
    return onPress().then(() => {
      expect(mockDispatch).toHaveBeenCalledWith({ type: 'STARTED_LOADING' });
      expect(mockDispatch).toHaveBeenCalledWith({ type: 'STOPPED_LOADING' });
      expect(mockCreateParentAuthentication).toHaveBeenCalled();
      expect(mockPersistParent).toHaveBeenCalled();
      expect(mockFetchReports).toHaveBeenCalled();
      expect(Actions.dashboard).toHaveBeenCalled();
    });
  });

  it('calls Alert when there is an error', () => {
    const testErrorMessage = 'Intentional test error - ignore!';
    const mockDispatch = jest.fn(() =>
      Promise.reject(new Error(testErrorMessage)));
    const { onPress } = mapDispatchToProps(mockDispatch);
    return onPress().then(() => {
      expect(mockDispatch).toHaveBeenCalledWith({ type: 'STARTED_LOADING' });
      expect(mockDispatch).toHaveBeenCalledWith({ type: 'STOPPED_LOADING' });
      expect(Alert.alert).toHaveBeenCalledWith(testErrorMessage);
    });
  });
});
