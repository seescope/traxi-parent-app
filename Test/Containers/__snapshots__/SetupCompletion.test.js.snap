// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameSetup Renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  style={
    Object {
      "backgroundColor": "#F0F2F7",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#545454",
            "fontFamily": "Raleway-ExtraBold",
            "fontSize": 26,
            "paddingHorizontal": 55,
            "textAlign": "center",
          }
        }
      >
        Last step!
      </Text>
      <View
        style={
          Object {
            "height": 16,
          }
        }
      />
      <View
        elevation={6}
        style={
          Array [
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 4,
              "margin": 8,
              "marginBottom": 24,
              "padding": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 3,
                "width": 3,
              },
              "shadowOpacity": 0.16,
              "shadowRadius": 3,
              "width": 734,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "marginVertical": 16,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#545454",
                "fontFamily": "Raleway-Regular",
                "fontSize": 14,
              }
            }
          >
            Your email address:
          </Text>
          <View>
            <TextInput
              autoCapitalize="words"
              autoCorrect={false}
              keyboardType="email-address"
              onChangeText={[Function]}
              onSubmitEditing={undefined}
              placeholderTextColor="rgba(75, 73, 73, 0.8)"
              secureTextEntry={undefined}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 4,
                  "color": "rgba(75, 73, 73, 1)",
                  "fontFamily": "Raleway-Regular",
                  "height": 32,
                  "marginTop": 4,
                  "paddingHorizontal": 8,
                  "width": 686,
                }
              }
              underlineColorAndroid="#448AFF"
              value="test@test.com"
            />
            <View
              style={
                Object {
                  "backgroundColor": "#448AFF",
                  "height": 2,
                  "marginBottom": 16,
                }
              }
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#545454",
                "fontFamily": "Raleway-Regular",
                "fontSize": 14,
              }
            }
          >
            Your name:
          </Text>
          <View>
            <TextInput
              autoCapitalize="words"
              autoCorrect={false}
              keyboardType={undefined}
              onChangeText={[Function]}
              onSubmitEditing={undefined}
              placeholderTextColor="rgba(75, 73, 73, 0.8)"
              secureTextEntry={undefined}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 4,
                  "color": "rgba(75, 73, 73, 1)",
                  "fontFamily": "Raleway-Regular",
                  "height": 32,
                  "marginTop": 4,
                  "paddingHorizontal": 8,
                  "width": 686,
                }
              }
              underlineColorAndroid="#448AFF"
              value={undefined}
            />
            <View
              style={
                Object {
                  "backgroundColor": "#448AFF",
                  "height": 2,
                  "marginBottom": 16,
                }
              }
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#545454",
                "fontFamily": "Raleway-Regular",
                "fontSize": 14,
              }
            }
          >
            Choose a password:
          </Text>
          <View>
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              keyboardType={undefined}
              onChangeText={[Function]}
              onSubmitEditing={undefined}
              placeholderTextColor="rgba(75, 73, 73, 0.8)"
              secureTextEntry={true}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 4,
                  "color": "rgba(75, 73, 73, 1)",
                  "fontFamily": "Raleway-Regular",
                  "height": 32,
                  "marginTop": 4,
                  "paddingHorizontal": 8,
                  "width": 686,
                }
              }
              underlineColorAndroid="#448AFF"
              value={undefined}
            />
            <View
              style={
                Object {
                  "backgroundColor": "#448AFF",
                  "height": 2,
                  "marginBottom": 16,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 16,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#448AFF",
              "fontFamily": "Raleway-Regular",
              "fontSize": 20,
            }
          }
        >
          See 
          Chris
          's usage
           
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
